### PR TITLE
Use forward slashes so logging commands do not fail in Linux

### LINKED ISSUE #3046: Async publishing from non-Windows platforms always fails during artifact creation
I attempted to on-board AspNetCore to publishing with Arcade, but immediately ran into this error message on all non-Windows platforms.

image
https://dev.azure.com/dnceng/internal/_build/results?buildId=223311&view=logs
```
##[error]Unable to process command '  ##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]/Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToPackageFeed.proj' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Path not exist: /Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToPackageFeed.proj
##[error]Unable to process command '  ##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]/Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToSymbolServers.proj' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Path not exist: /Users/vsts/agent/2.152.1/work/1/s/eng/common\PublishToSymbolServers.proj
##[error]Unable to process command '  ##vso[artifact.upload containerfolder=ReleaseConfigs;artifactname=ReleaseConfigs]/Users/vsts/agent/2.152.1/work/1/s/eng/common\CheckSymbols.ps1' successfully. Please reference documentation (http://go.microsoft.com/fwlink/?LinkId=817296)
##[error]Path not exist: /Users/vsts/agent/2.152.1/work/1/s/eng/common\CheckSymbols.ps1
```

My guess: this would be fixed by normalizing file paths to use forward slashes on Linux/macOS.

cc @JunTaoLuo @markwilkie

### PR DESCRIPTION
**Pull request by riarenas**
Fixes #3046

### CODE REVIEW

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj [hunk 1]
```diff
@@ -178,10 +178,10 @@
 
     <Copy 
       SourceFiles="$(AssetManifestFilePath)" 
-      DestinationFolder="$(TempWorkingDirectory)\$(AssetManifestFileName)" />
+      DestinationFolder="$(TempWorkingDirectory)/$(AssetManifestFileName)" />
     
     <Message
-      Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)\$(AssetManifestFileName)"
+      Text="##vso[artifact.upload containerfolder=AssetManifests;artifactname=AssetManifests]$(TempWorkingDirectory)/$(AssetManifestFileName)"
```
Comment (natemcmaster) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj]: This should work.

Tip, if you didn't know already, you can also have done`$([MSBuild]::NormalizePath($(TempWorkingDirectory), $(AssetManifestFileName)))` to let MSBuild figure out the right slashes to use.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj [hunk 2]
```diff
@@ -26,9 +26,9 @@
     <PublishToSymbolServer Condition="'$(UsingToolSymbolUploader)' == 'true' and '$(AzureFeedUrl)' == '' and '$(ContinuousIntegrationBuild)' == 'true'">true</PublishToSymbolServer>
 
     <AssetManifestFileName>$(OS)-$(PlatformName).xml</AssetManifestFileName>
-    <AssetManifestFilePath>$(ArtifactsLogDir)AssetManifest\$(AssetManifestFileName)</AssetManifestFilePath>
+    <AssetManifestFilePath>$(ArtifactsLogDir)AssetManifest/$(AssetManifestFileName)</AssetManifestFilePath>
```
Comment (natemcmaster) [inline src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj]: Our coding style in MSBuild files in Arcade is to use backslashes everywhere. Usually MSBuild normalizes these automatically, but it appears it doesn't normalize inputs to the `<Message>` task.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj (hunk 1 above)
Comment (riarenas) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj]: Oh. Didn't know that. Thanks for the tip.

File: src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj (hunk 2 above)
Comment (riarenas) [reply src/Microsoft.DotNet.Arcade.Sdk/tools/Publish.proj]: Gotcha. I'm reverting these and only changing the ones in the logging commands.

**Pull request merged**